tests/cmsis_rtos_v1/src/thread_apis.c

```diff
@@ -106,11 +108,22 @@ void test_thread_prio(void)
 
 	status = osThreadTerminate(id3);
 	zassert_true(status == osOK, "Error terminating thread3");
+
+	/* Try to set unsupported priority and assert failure */
+	status = osThreadSetPriority(osThreadGetId(), osPriorityDeadline);
+	zassert_false(status == osOK, "Invalid priority set");
+
+	/* Try to set priority to inactive thread and assert failure */
+	status = osThreadSetPriority(id3, osPriorityNormal);
+	zassert_false(status == osOK,
+			"Priority cannot be set to inactive thread");
```

Comment:
Check for osErrorResource.  The statement can be something like
zassert_true(status == osErrorResource, "Something's wrong with osThreadSetPriority!");